docs/zkapps/tutorials/09-recursion.mdx

```diff
@@ -56,86 +56,310 @@ You build recursive zkApps with [ZkProgram](/zkapps/o1js-reference/modules/Exper
 
 Proofs generated using a ZkProgram can be passed into zkApp smart contracts for them to verify recursively. They can even be passed recursively into their own functions for off-chain recursive composition.
 
-The following ZkProgram example code is provided in the [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/09-recursion/src/main.ts) example file.
+The full source code for the following ZkProgram tutorial is provided in the [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/09-recursion/src/main.ts) directory on GitHub. While you're there, give the `/docs2` repository a star so that other zk developers can learn to build a zkApp!
 
-To create a ZkProgram, start with the `init()` method. 
+:::info
 
-- For each method, declare the inputs it will receive. 
-- The first argument of a ZkProgram method is always the state of the ZkProgram, named `publicInput` since it is public.
+To prevent copying line numbers and command prompts as shown in the examples, use the copy code to clipboard button that appears at the top right of the snippet box when you hover over it.
 
-```typescript
-const Add = ZkProgram({
-  name: 'add-example',
-  publicInput: Field,
+:::
 
-  methods: {
-    init: {
-      privateInputs: [],
+## Prerequisites
+
+Ensure your environment meets the [Prerequisites](/zkapps/tutorials#prerequisites) for zkApp Developer Tutorials.
+
+In particular, make sure you have the zkApp CLI installed:
+
+```sh
+$ npm install -g zkapp-cli
+```
+
+## Create a new project
+
+Now that you have the tooling installed, you can start building your application.
+
+1. Create or change to a directory where you have write privileges.
+1. Now, create a project using the `zk project` command:
+
+```sh
+$ zk project 09-recursion
+```
+
+The `zk project` command has the ability to scaffold the UI for your project. For this tutorial, select `none`:
+
+```
+? Create an accompanying UI project too? …
+  next
+  svelte
+  nuxt
+  empty
+❯ none
+```
+
+The expected output is:
+
+```sh
+✔ Create an accompanying UI project too? · none
+✔ UI: Set up project
+✔ Initialize Git repo
+✔ Set up project
+✔ NPM install
+✔ NPM build contract
+✔ Set project name
+✔ Git init commit
+
+Success!
```

Comment:
```suggestion
The `zk project` command has the ability to scaffold the UI for your project. For this tutorial, select `none`.
```

Reply:
since this is tutorial 9, we can leave out the very detailed steps since devs will have already mastered these steps in earlier tutorials